src/library/dao/autodjcratesdao.cpp

```diff
@@ -881,3 +886,81 @@ void AutoDJCratesDAO::slotPlayerInfoTrackUnloaded(QString group,
         }
     }
 }
+//
+// We are selecting the track in the following manner.
+// We divide the library tracks into two section . For which
+// we sort the library according to times_played and select a
+// percentage_of_prefered_tracks (25% by default) .
+//
+// What we do then is either select a random track from this %25
+// or a random track from the remaining 75% both with equal
+// probability .
+//
+// (Assuming that 25% tracks are unplayed or very rarely played)
+//
+// This ensures that there is a 50% chance that an rarely played
+// track is returned .
+//
+// This is better than returning the least played track because
+// in that case there may be instances where the same track is
+// returned by this function on multiple successive calls which is
+// not desired.
+//
+// Further this also does not restrict our function to only retrieve
+// unplayed tracks (there is probably a reason they are unplayed).
+///
+int AutoDJCratesDAO::getRandomTrackIdFromLibrary(void) {
```

Comment:
Just an Idea:
If we assume that th garbage tracks have a play counter of 0 and the heavy rotation tracks have a high playcount. 
Your algorithm should prefer tracks between. What about just picking a window from 25% to 75% and toogle it with a 100 % query?